src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -9,11 +9,17 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Hosting.Internal
 {
     public class HostingApplication : IHttpApplication<HostingApplication.Context>
     {
+#if !NET451
```

Comment:
@lmolkova  I think we have now made DiagnosticSource work on V4.5 right?     (that is what the Activity.Current.net45.cs is about right?)   If so we should not need #If defs around this.    We really should figure this out and get rid of them.